src/battle_ai_switch_items.c

```diff
@@ -977,7 +977,7 @@ u8 GetMostSuitableMonToSwitchInto(void)
         }
     }
 
-    bestMonId = GetBestMonBatonPass(party, firstId, lastId, invalidMons, aliveCount);
+    bestMonId = aliveCount == 1 ? PARTY_SIZE : GetBestMonBatonPass(party, firstId, lastId, invalidMons, aliveCount);
```

Comment:
I don't think this is the part that's fixing the issue. Passing `aliveCount` as 1 would always return `PARTY_SIZE` always, because the `return i;` is only ever reachable if `aliveCount` is 2 or more.

Reply:
Yeah, I don't know what I I thought this will accomplish.